emscripten.py

```diff
@@ -908,8 +908,8 @@ def report_missing_symbols(all_implemented, pre):
       continue
     diagnostics.warning('undefined', 'undefined exported function: "%s"', requested)
 
-  # Handle main specially, unless IGNORE_MISSING_MAIN is set
-  if shared.Settings.EXPECT_MAIN and '_main' not in all_implemented and not shared.Settings.IGNORE_MISSING_MAIN:
+  # Handle main specially, unless IGNORE_MISSING_MAIN or STANDALONE_WASM mode
+  if not shared.Settings.STANDALONE_WASM and shared.Settings.EXPECT_MAIN and '_main' not in all_implemented and not shared.Settings.IGNORE_MISSING_MAIN:
```

Comment:
could we set `IGNORE_MISSING_MAIN` in standalone mode?

Reply:
I think that these represent two different reasons for not reporting the error here.

1.  IGNORE_MISSING_MAIN is set meaning that we allow the build to compile even if they have not main function.  In this case we obviously don't want to error out.
2. STANDALONE_WASM we do require a main entry point but (a) its called _start and (b) we fail at link time if its missing.

Reply:
Maybe I will add some more docs here.